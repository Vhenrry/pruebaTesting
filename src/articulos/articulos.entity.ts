export class Articulo {
  id: string
  titulo: string
  contenido: string
  categoria: string
  _fecha_creacion: Date
}