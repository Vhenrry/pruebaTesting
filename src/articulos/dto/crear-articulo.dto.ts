export class CrearArticuloDto {
  readonly titulo: string
  readonly contenido: string
  readonly categoria: string
}