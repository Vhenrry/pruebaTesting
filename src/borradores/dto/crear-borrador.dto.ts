export class CrearBorradorDto {
  readonly titulo: string
  readonly categoria: string
}