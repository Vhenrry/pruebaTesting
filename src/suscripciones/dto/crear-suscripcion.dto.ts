export class CrearSuscripcionDto {
  readonly idUsuario: string
  readonly idCategoria: string
}