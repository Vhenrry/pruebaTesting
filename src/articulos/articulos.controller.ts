import { Body, Controller, Delete, Get, Param, Post, Put } from '@nestjs/common';
import { ArticulosService } from './articulos.service';
import { CrearArticuloDto } from './dto/crear-articulo.dto';
import { Articulo } from './articulos.entity';

@Controller('articulos')
export class ArticulosController {
  constructor(private readonly articulosService: ArticulosService){}

  @Post()
  crearArticulo(@Body() crearArticuloDto: CrearArticuloDto): Articulo {
    return this.articulosService.crear(crearArticuloDto)
  }

  @Get()
  buscarArticulos(): Articulo[] {
    return this.articulosService.buscarTodo()
  }

  @Put(':id')
  modificarArticulo(@Param('id') id: string, @Body() actualizarArticuloDto: Partial<CrearArticuloDto>): Articulo {
    return this.articulosService.modificar(id, actualizarArticuloDto)
  }
  @Delete(':id')
  eliminarArticulo(@Param('id') id: string): void {
    return this.articulosService.eliminar(id)
  }

}
