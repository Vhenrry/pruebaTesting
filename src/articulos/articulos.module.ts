import { Module } from '@nestjs/common';
import { ArticulosController } from './articulos.controller';
import { ArticulosService } from './articulos.service';
import { ArticulosRepository } from './articulos.repository';

@Module({
  controllers: [ArticulosController],
  providers: [ArticulosService, ArticulosRepository]
})
export class ArticulosModule {}
