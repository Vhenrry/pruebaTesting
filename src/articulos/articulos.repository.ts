import { Injectable, NotFoundException } from '@nestjs/common'
import { Articulo } from './articulos.entity'
import { CrearArticuloDto } from './dto/crear-articulo.dto'


@Injectable()
export class ArticulosRepository {
  private articulos: Articulo[] = []

  crear(crearArticuloDto: CrearArticuloDto): Articulo {

    const nuevoArticulo: Articulo = {
      id: (this.articulos.length +1).toString(),
      ...crearArticuloDto,
      _fecha_creacion: new Date()
    }

    this.articulos.push(nuevoArticulo)
    return nuevoArticulo
  }

  buscar(): Articulo[] {
    return this.articulos
  }

  buscarPorId(id: string): Articulo {
    const articulo =  this.articulos.find(itemArticulo => itemArticulo.id === id)
    if (!articulo) {
      throw new NotFoundException('No se encontró el artículo buscado')
    }

    return articulo
  }

  actualizar(id: string, articuloActualizarDto: Partial<CrearArticuloDto>): Articulo {
    const articulo =  this.buscarPorId(id)
    Object.assign(articulo, articuloActualizarDto)
    return articulo
  }

  eliminar(id: string): void {
    const indiceArticulo = this.articulos.findIndex(articulo => articulo.id === id)
    if (indiceArticulo === -1 ){
      throw new NotFoundException('No se encontró el artículo a eliminar')
    }
    this.articulos.splice(indiceArticulo, 1)
  }

}