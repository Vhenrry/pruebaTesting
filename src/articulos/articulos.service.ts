import { Injectable } from '@nestjs/common';
import { ArticulosRepository } from './articulos.repository';
import { CrearArticuloDto } from './dto/crear-articulo.dto';
import { Articulo } from './articulos.entity';

@Injectable()
export class ArticulosService {
  constructor(private readonly articulosRepository: ArticulosRepository) {}

  crear(crearArticuloDto: CrearArticuloDto ): Articulo {
    return this.articulosRepository.crear(crearArticuloDto)
  }

  buscarTodo(): Articulo[] {
    return this.articulosRepository.buscar()
  }

  buscarPorId(id: string): Articulo {
    return this.articulosRepository.buscarPorId(id)
  }

  modificar(id: string, articuloActualizarDto:Partial<CrearArticuloDto>): Articulo {
    return this.articulosRepository.actualizar(id, articuloActualizarDto)
  }

  eliminar(id: string): void {
    this.articulosRepository.eliminar(id)
  }

}
