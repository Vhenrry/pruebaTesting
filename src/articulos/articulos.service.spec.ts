import { Test, TestingModule } from '@nestjs/testing';
import { ArticulosService } from './articulos.service';
import { CrearArticuloDto } from './dto/crear-articulo.dto';
import { Articulo } from './articulos.entity';
import { ArticulosRepository } from './articulos.repository';

describe('ArticulosService', () => {
  let service: ArticulosService;
  let repository: ArticulosRepository

  beforeEach(async () => {
    const module: TestingModule = await Test.createTestingModule({
      providers: [ArticulosService,ArticulosRepository],
    }).compile();

    service = module.get<ArticulosService>(ArticulosService);
    repository = module.get<ArticulosRepository>(ArticulosRepository);
  });

  it('should be defined', () => {
    expect(service).toBeDefined();
  });

  it('deberia crear un articulo', () => {
    const miArticulo: Articulo = new Articulo()
    miArticulo.id= '1',
    miArticulo.titulo= 'Mi primer articulo',
    miArticulo.contenido= 'Las pruebas unitarias son la clave',
    miArticulo.categoria= 'Medicina'
    jest.spyOn(repository, 'crear').mockImplementation(() => miArticulo)
    const nuevoArticulo: CrearArticuloDto = {
      titulo: 'Mi primer articulo',
      contenido: 'Las pruebas unitarias son la clave',
      categoria: 'Medicina'
    }
    const respuesta = service.crear(nuevoArticulo)
    expect(respuesta).toBeInstanceOf(Articulo)
  })

  it('deberia modificar un articulo', () => {
    const miArticulo: Articulo = new Articulo()
    
    miArticulo.id= '1',
    miArticulo.titulo= 'Mi primer articulo',
    miArticulo.contenido= 'Las pruebas unitarias son la clave',
    miArticulo.categoria= 'Medicina'

    
    jest.spyOn(repository, 'actualizar').mockImplementation(() => miArticulo)
    const nuevoArticulo: Partial<CrearArticuloDto> = {
      titulo: 'Mi primer articulo modificado',
    }
    const resultado = service.modificar('1', nuevoArticulo)
    expect(resultado).toBeInstanceOf(Articulo)
  })

  it('deberia devolver todos los articulos registrados', async () => {
    const findAllSpy = jest.spyOn(repository, 'buscar');
    await service.buscarTodo();
    expect(findAllSpy).toHaveBeenCalledTimes(1);
  });

  it('should create a new articulo', async () => {
    const articulo = new Articulo();
    const result = await service.crear(articulo);
    expect(result).toEqual(expect.objectContaining(articulo));
  });

  it('should find all articulos', async () => {
    const result = await service.buscarTodo();
    expect(result).toBeInstanceOf(Array);
  });

});
