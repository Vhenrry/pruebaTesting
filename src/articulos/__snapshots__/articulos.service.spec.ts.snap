// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticulosService should delete an articulo: No se encontró el artículo buscado 1`] = `"No se encontró el artículo a eliminar"`;
